TAScheduler/UnitTests/tests.py

```diff
@@ -46,38 +52,476 @@ class TestAdminCreateSection(TestCase):  # Alec
+            email_address='admin@example.com',
+            password='adminpassword',
+            first_name='Admin',
+            last_name='User',
+            home_address='123 Admin St',
+            phone_number='1234567890'
+        )
+        hold_user.save()
+        hold_admin = Administrator(user=hold_user)
+        hold_admin.save()
+        self.admin = AdminObj(hold_admin)
+
+    def test_bad_instr(self):
+        with self.assertRaises(TypeError, msg='TA that was passed is not a valid TA'):
+            self.admin.courseInstrAsmgt(11, self.tempCourse)
+
+    def test_bad_course(self):
+        with self.assertRaises(TypeError, msg='Course that was passed is not a valid course'):
+            self.admin.courseInstrAsmgt(self.tempTA, 11)
+
+    def test_null_instr(self):
+        User.delete(self.hold_user)
+        with self.assertRaises(RuntimeError, msg="Tried to link course to non existant user"):
+            self.admin.courseTAAsmgt(self.tempTA, self.tempCourse)
+
+    def test_null_course(self):
+        Course.delete(self.hold_course)
+        with self.assertRaises(RuntimeError, msg="Tried to link user to non existant course"):
+            self.admin.courseTAAsmgt(self.tempTA, self.tempCourse)
+
+    def test_success_connect(self):
+        self.admin.courseTAAsmgt(self.tempTA, self.tempCourse)
+        self.assertEqual(TAToCourse.objects.get(course=self.hold_course).course,
+                         self.hold_course, "Should have linked course and TA together")
+
+    def test_max_capacity_instr(self):
+        User.delete(self.hold_user)
+        temp = User.objects.create(self.hold_user)
+        ta = TA.objects.create(
+            user=temp,
+            max_assignments=0,
+            grader_status=False
+        )
+        temp_ta = TAObj(ta)
+        with self.assertRaises(RuntimeError, msg="Tried to link course to TA with max assignments"):
+            self.admin.courseTAAsmgt(temp_ta, self.tempCourse)
+
 
 class TestTAInit(TestCase):
-    pass
+    ta_database = None
+    user = None
+
+    def setUp(self):
+        self.user = User.objects.create(
+            email_address='admin@example.com',
+            password='adminpassword',
+            first_name='Admin',
+            last_name='User',
+            home_address='123 Admin St',
+            phone_number='1234567890'
+        )
+        self.ta_database = TA.objects.create(user=self.user, grader_status=False)
+
+    def test_bad_input(self):
+        with self.assertRaises(TypeError, msg='TA that was passed is not a valid TA'):
+            ta = TAObj(11)
+
+    def test_null_ta(self):
+        User.delete(self.user)
+        with self.assertRaises(TypeError, msg='TA that was passed does not exist'):
+            ta = TAObj(self.ta_database)
+
+    def test_success(self):
+        ta = TAObj(self.ta_database)
+        self.assertEqual(ta.databasereference, self.ta_database,
+                         "TA object should be saved in the database reference")
```

Comment:
I don't know if you meant to do this without calling the __init__() method or if what you did is the same as calling it, but I think you need the __init__() method

Reply:
You actually don't what I am doing is implicitly calling the init function